LunarVim/LunarVim
# refactor: update impatient

# Description

- update impatient: merge the the latest upstream version.  _it still needs to be loaded before any plugins, so it will have to be managed manually either way_
- feat: use an independent cache directory: overridden `vim.fn.stdpath` to use LUNARVIM_x_DIR instead, since a lot of plugins call this function interally

resolves #1256
fixes #2490

## How to test

- added new tests for `vim.fn.stdpath`: 
https://github.com/LunarVim/LunarVim/blob/211fa202640932844a886553e6d5f5733002442d/tests/specs/bootstrap_spec.lua#L13-L22
- check https://github.com/LunarVim/LunarVim/pull/2477#issuecomment-1102702593


## Notes

For future references: 
- it's easy to restore the original behavior of `vim.fn.stdpath`, by either removing some entries from https://github.com/kylo252/LunarVim/blob/978820b8edc56a12fd0c34664d6adabbe4e51734/lua/lvim/bootstrap.lua#L73-L83, or overriding the function, yet again, in `config.lua`
